nixos/modules/services/networking/https-dns-proxy.nix

```diff
@@ -0,0 +1,86 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.https-dns-proxy;
+
+  providerCfg =
+    if cfg.provider == "cloudflare"
```

Comment:
I am against hard coding this. 
Also this limits support to ipv4.
Can we just leave this open? 
I for example run `https://dns.thalheim.io/dns-query`.
In an option an example for cloudflare or google could be listed though.
However I would leave it up to to the user to make a choice explicitly.

Reply:
How about a `custom` option? So cf, google or custom and you then need to provide the servers yourself?

Reply:
Instead of using `custom`, it's IMHO better to just make `provider` and (something like) `dnsUrl` mutually exclusive options.

Reply:
I would not add any provider option at all. I think users should educate themselves, where they sent all their DNS queries too. I would rather link to https://github.com/curl/curl/wiki/DNS-over-HTTPS#publicly-available-servers in the documentation.